lib/inc/hocon/config_value.hpp

```diff
@@ -52,6 +45,25 @@ namespace hocon {
         friend class config_concatenation;
     public:
         /**
+         * The type of a configuration value (following the <a
+         * href="http://json.org">JSON</a> type schema).
+         */
+        enum class type {
+            OBJECT, LIST, NUMBER, BOOLEAN, CONFIG_NULL, STRING, UNSPECIFIED
+        };
+        static char const* type_name(type t) {
```

Comment:
GCC 5.3 is throwing a "control reaches end of non-void function" warning here for me. Obviously that can't happen if `t` is a valid value. If I add a default that throws a `std::logic_error("Got impossible value for type enum")` it compiles